project/Dependencies.scala

```diff
@@ -99,6 +99,7 @@ object Dependencies {
     val jacksonAfterburner = "com.fasterxml.jackson.module" % "jackson-module-afterburner" % jacksonVersion // ApacheV2
     val jacksonCbor = "com.fasterxml.jackson.dataformat" % "jackson-dataformat-cbor" % jacksonVersion // ApacheV2
     val jacksonSmile = "com.fasterxml.jackson.dataformat" % "jackson-dataformat-smile" % jacksonVersion // ApacheV2
+    val lz4_java = "org.lz4" % "lz4-java" % "1.6.0"
```

Comment:
rename to `lz4Java` or just `lz4`

Reply:
add `// ApacheV2` note at the end

Reply:
It has zero transitive dependencies, good.
One consideration is if we should make lz4-java an optional dependency, but on the other hand if compression is to be used there it's better to use lz4 than gzip. Maybe we should make lz4 the default compression algorithm.
I'm thinking that perhaps we should make the compressed versions separate serializers so we have bindings:
* `jackson-json` - no compression
* `jackson-cbor` - no compression
* `jackson-json-lz4` - lz4 compression
* `jackson-cbor-lz4` - lz4 compression

For gzip it can optionally be enabled for the plain `jackson-json` since it has the magic (we mostly need to support gzip for backwards compatibility with Lagom).

I could do this part as a separate PR afterwards if you like.

Reply:
Or we could add our own magic prefix for lz4. Perhaps that is better?

Reply:
Renamed to `lz4Java` and added Apache license note.
I cannot oversee the impact of changing the default compression to lz4. Would that not be a breaking change? Could it be something would break when doing a rolling update or when communicating with external services or db?
Separate serialisers seems a good idea. Would you do that? I think you'd be way faster than me.

Adding own magic prefix would not be compatible with non-akka services sending or expecting a normal LZ4 byte stream, so I would not recommend that.

Reply:
Looks like there is a magic in the specification, and lz4-java has support for it: https://github.com/lz4/lz4-java/blob/master/src/java/net/jpountz/lz4/LZ4FrameOutputStream.java

Reply:
Mmm ... indeed. It was not mentioned in the LZ4 spec, but I'll have a look.